lib/util/mailables.js

```diff
@@ -0,0 +1,154 @@
+import blobStream from 'blob-stream'
+import moment from 'moment'
+
+import PDFDocument from './PDFDocumentWithTables'
+
+export const MAILABLE_FIELDS = [
+  {fieldName: 'firstname', placeholder: 'First name'},
+  {fieldName: 'lastname', placeholder: 'Last name'},
+  {fieldName: 'address1', placeholder: 'Address 1'},
+  {fieldName: 'address2', placeholder: 'Address 2'},
+  {fieldName: 'city', placeholder: 'City'},
+  {fieldName: 'state', placeholder: 'State'},
+  {fieldName: 'zip', placeholder: 'Zip'}
+]
+
+export function createLetter (formData, mailablesConfig) {
+  // This is a very goofy approach to convert an image URL to its image data for
+  // writing to the PDF, but it seems to be a solid approach.
+  const img = new Image()
+  img.crossOrigin = 'anonymous'
+  img.onload = () => {
+    var canvas = document.createElement('canvas')
+    canvas.width = img.width
+    canvas.height = img.height
+
+    const ctx = canvas.getContext('2d')
+    ctx.drawImage(img, 0, 0)
+
+    const data = canvas.toDataURL('image/png')
+    const imageData = {data, height: img.height, width: img.width}
+    writePDF(formData, imageData, mailablesConfig)
+  }
+  img.src = mailablesConfig.header_graphic
+}
+
+function writePDF (formData, imageData, otpConfig) {
+  const {
+    address1 = '',
+    address2,
+    city = '',
+    firstname = '',
+    lastname = '',
+    mailables,
+    state = '',
+    zip = ''
+  } = formData
+  const {
+    horizontal_margin: horizontalMargin,
+    vertical_margin: verticalMargin
+  } = otpConfig
+  const margins = {
+    top: verticalMargin,
+    bottom: verticalMargin,
+    left: horizontalMargin,
+    right: horizontalMargin
+  }
+  console.log(margins)
+  const doc = new PDFDocument({margins})
+  const stream = doc.pipe(blobStream())
+  preparePage(doc, imageData, otpConfig)
+  doc.on('pageAdded', () => preparePage(doc, otpConfig))
```

Comment:
This seems to be called with a different set of arguments than the line above. Is this needed?

Reply:
I think ya found a bug. Thanks!